agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,310 +95,328 @@ public CallReplacementTransformer() {}
    */
-  public void instrument_method(
-      Method m, MethodGen mg, Map<MethodDef, ReplacementClass> current_method_map) {
-
+  private boolean transformMethod(MethodGen mg, InstructionFactory ifact, ConstantPoolGen pgen) {
+    boolean transformed = false;
     // Loop through each instruction, making substitutions
-    InstructionList il = mg.getInstructionList();
-    for (InstructionHandle ih = il.getStart(); ih != null; ) {
-      if (debug_instrument_inst.enabled()) {
-        debug_instrument_inst.log("instrumenting instruction %s%n", ih);
-        // ih.getInstruction().toString(pool.getConstantPool()));
+    debug_transform.log("transformMethod: ENTER %s.%s%n", mg.getClassName(), mg.getName());
+    InstructionList instructionList = mg.getInstructionList();
+    for (InstructionHandle instructionHandle = instructionList.getStart();
+        instructionHandle != null;
+        ) {
+
+      debug_instrument_inst.log(
+          "%s.%s: instrumenting instruction %s%n",
+          mg.getClassName(), mg.getName(), instructionHandle);
+
+      // The next instruction for next iteration
+      InstructionHandle nextHandle = instructionHandle.getNext();
+
+      InvokeInstruction instruction =
+          getReplacementInstruction(mg, instructionHandle.getInstruction(), ifact, pgen);
+      debug_instrument_inst.log(
+          " %s.%s new inst: %s%n", mg.getClassName(), mg.getName(), instruction);
+
+      if (instruction != null) {
+        transformed = true;
+        replaceInstructions(instructionList, instructionHandle, instruction);
       }
 
-      // Remember the next instruction to process
-      InstructionHandle next_ih = ih.getNext();
-
-      // Get the translation for this instruction (if any)
-      InstructionList new_il = xform_inst(mg, ih.getInstruction(), current_method_map);
-      if (debug_instrument_inst.enabled()) debug_instrument_inst.log("  new inst: %s%n", new_il);
-
-      // If this instruction was modified, replace it with the new
-      // instruction list. If this instruction was the target of any
-      // jumps or line numbers, replace them with the first
-      // instruction in the new list.
-      replace_instructions(il, ih, new_il);
-
-      ih = next_ih;
+      instructionHandle = nextHandle;
     }
+    debug_transform.log("transformMethod: EXIT %s.%s%n", mg.getClassName(), mg.getName());
+    return transformed;
   }
 
   /**
-   * Transforms invoke instructions that match the specified list for this class to call the
-   * specified static call instead.
+   * Transforms invoke instructions that match a replacement so that it calls the replacement method
+   * instead of the original method.
+   *
+   * @param mg the BCEL representation of the method being transformed
+   * @param inst the instruction to transform
+   * @param ifact the instruction factory for the enclosing class
+   * @return the transformed instruction list, or null if the instruction is not transformed
```

Comment:
This returns a single instruction, not an instruction list.  There are a few other errors of this variety elsewhere.